src/lib/Siteaccess/RootLocationsProvider.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Siteaccess;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+
+class RootLocationsProvider
+{
+    private const ROOT_LOCATIONS_PARAMETER_NAMES = [
+        'content.tree_root.location_id',
+        'location_ids.media',
+        'form_builder.forms_location_id',
```

Comment:
You can't do it this way - AdminUI doesn't (or shouldn't) know about `ezplatform-form-builder` package. There is no dependency in this direction but there is opposite dependency which means you can extend AdminUI in FormBuilder. You can create an extension point in `ContentController::previewAction` to run decision voters. At least this is how I see the implementation of it.